js/components/graph/Node.js

```diff
@@ -2,57 +2,99 @@ import PropTypes from "prop-types";
 export default class Node extends React.Component {
+  /** Create a node */
   constructor(props) {
     super(props);
     var state = localStorage.getItem(this.props.JSON.id_);
+
+    /**
+     * Unselected states:
+     *  - takeable: all prerequisites are satisfied (or no prereqs/parents)
+     *  - inactive: missing some prerequisites
+     * 
+     * Selected states:
+     *  - active: all prerequisites are satisfied
+     *  - overridden: missing some prerequisites (will have a red border)
+     */
+
     if (this.props.editMode) {
       state = ""; // TODO: define what editMode is
     } else if (state === null) {
+      // sets state to takeable if it has no parent nodes (prereqs)
       state = this.props.parents.length === 0 ? "takeable" : "inactive";
     }
+
     this.state = {
       status: state,
-      selected: ["active", "overridden"].indexOf(state) >= 0
+      selected: ["active", "overridden"].indexOf(state) >= 0 // checks if state means its selected
     };
   }
 
+  /**
+   * Checks whether this Node is selected
+   * @return {boolean}
+   */
   isSelected = () => {
+    // Hybrid nodes: the grey, smaller nodes representing another node farther way
+    // They can only be either active or inactive.
     if (this.props.hybrid) {
       return this.state.status === "active";
     } else {
       return this.state.selected;
     }
   }
 
+  /**
+   * Checks whether all prerequisite/ preceeding nodes for the current one are satisfied
+   * @return {boolean}
+   */
   arePrereqsSatisfied = () => {
-    var svg = this.props.svg;
+    var svg = this.props.svg;  // contains information about the whole graph
+
+    /**
+     * Recursively checks that preceeding nodes are selected
+     * @param  {string|Array} element Node(s)/ other on the graph
+     * @return {boolean}
+     */
     function isAllTrue(element) {
+      // base case: single node
       if (typeof element === "string") {
+        // if the element represents a defined node, return whether it is a selected
         if (svg.nodes.current[element] !== undefined) {
           return svg.nodes.current[element].isSelected();
-        } else if (svg.bools.current[element] !== undefined) {
+        } else if (svg.bools.current[element] !== undefined) { // else if it is a defined Bool
           return svg.bools.current[element].isSelected();
         } else {
           return false;
         }
       } else {
+        // check if at least one node is selected from element
         return element.some(isAllTrue);
       }
     }
 
+    // checks if parent nodes are selected
     return this.props.parents.every(isAllTrue);
   }
 
+  /**
+   * Update the state/status of a node (and its children/edges)
+   * @param  {boolean} recursive whether we should recurse on its children
+   */
   updateNode = recursive => {
     var newState;
+
+    // if the prereqs are satisfied, the node is takeable. Once selected it becomes active.
```

Comment:
Remove the inline comments in this function (same comment as above).

Reply:
The one on line 107 was not added by me, should I keep it?

Reply:
Good question! That's an example of a comment which isn't adding much, but could be improved to explain the *why* of the condition. So why don't you change it to a comment like "We can stop without updating the children when the selected state of this node hasn't changed and the original state wasn't "missing"."